dfaker.py

```diff
@@ -0,0 +1,387 @@
+	bolus, wizard = bolus_or_wizard(cleaned)
+	np_bolus, np_wizard = np.array(bolus), np.array(wizard)
+	b_carbs, b_ts  = np_bolus[:, 0], np_bolus[:, 1]
+	w_card, w_ts, w_gluc = np_wizard[:, 0], np_wizard[:, 1], np_wizard[:,2]
+	return b_carbs, b_ts, w_card, w_ts, w_gluc
+
+def clean_up_boluses(carb_time, filter_rate=7):
+	""" Cleans up clusters of carb events based on meal freqeuncy
+		carb_time -- a numpy array of carbohydrates at different points in time
+		filter_rate -- how many 5 minute segments should be filtered within the range
+					   of a single event.
+					   Example: 7 * 5 --> remove the next 35 minutes of data from carb_time
+	"""
+	return carb_time[::filter_rate]
+
+def remove_night_boluses(carb_time_gluc):
+	"""Removes night boluses excpet for events with high glucose""" 
+	keep = []
+	for row in carb_time_gluc:
+		hour = time.gmtime(row[1]).tm_hour
+		if (hour > 6 and hour < 23):
+			keep.append(row)
+		elif int(row[2]) not in range(0,250):
+			keep.append(row)
+	np_keep = np.array(keep)
+	return np_keep
+
+def bolus_or_wizard(solution):
+	bolus_events, wizard_events = [], []
+	for row in solution:
+		bolus_or_wizard = random.randint(0, 5)
+		if bolus_or_wizard == 2 or bolus_or_wizard == 4:
+			bolus_events.append(row)
+		else:
+			wizard_events.append(row)
+	return bolus_events, wizard_events
+
+def convert_to_mmol(iterable):
+	conversion_factor = 18.01559
+	return [reading / conversion_factor for reading in iterable]
+
+def make_timesteps(start_timestamp, timelist):
+	""" Convert list of floats representing time into epoch time"""
+	timesteps = []
+	epoch_ts = start_timestamp.strftime('%s')
+	for time_item in timelist:
+		new_time = int(epoch_ts) + int(time_item * 60) 
+		timesteps.append(new_time)
+	return timesteps
+
+def cbg(gluc, timesteps, params):
+	for value, timestamp in zip(convert_to_mmol(gluc), timesteps):
+		cbg_reading = {}
+		cbg_reading["deviceId"] = "MiniMed 530G - 551-=-53875424"
+		cbg_reading["id"] = str(uuid.uuid4())
+		cbg_reading["uploadId"] = "upid_fdbde582fe2b"
+		cbg_reading["type"] = "cbg"
+		cbg_reading["value"] = value
+		cbg_reading["deviceTime"] = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime(timestamp))
+		cbg_reading["timezoneOffset"] = get_offset(params)
+		cbg_reading["time"] = (time.strftime('%Y-%m-%dT%H:%M:%S.000Z', time.gmtime(timestamp 
+								- cbg_reading["timezoneOffset"]*60)))
+		cbg_reading["units"] = "mmol/L"
+		dfaker.append(cbg_reading)
+
+def smbg(gluc, timesteps, params):
+	fingersticks_per_day = params['smbg_freq']
+	total = (24 * 60) / 5
+	increment = int(total / fingersticks_per_day)
+	time_gluc = list(zip(convert_to_mmol(gluc), timesteps))
+	for value, timestamp in time_gluc[::increment]:
+		smbg_reading = {}
+		smbg_reading["deviceId"] = "MiniMed 530G - 551-=-53875424"
+		smbg_reading["id"] = str(uuid.uuid4())
+		smbg_reading["type"] = "smbg"
+		smbg_reading["uploadId"] = "upid_fdbde582fe2b"
+		smbg_reading["value"] = value + random.uniform(-1.5, 1.5) #in mmol/L
```

Comment:
I know this is happening, and the reason is that I was shifting the time as well, adding a randomize_time to device time. I did that so that the smbg entries are not evenly distributed, but the result is that at times they are shifted more than 1.5 units.